openstf/pipeline/train_create_forecast_backtest.py

```diff
@@ -69,3 +205,38 @@ def train_model_and_forecast_back_test(
     forecast["horizon"] = test_data.iloc[:, -1]
 
     return forecast, model, train_data, validation_data, test_data
+
+
+def group_kfold(
```

Comment:
Nice function, it seems quite general. Is this the best place to declare it? Maybe it is better to define it in the part that also houses the splitting functionality?